src/TR_alg.jl

```diff
@@ -186,8 +186,9 @@ function TR(
     set_radius!(ψ, ∆_effective)
     subsolver_options.Δk = ∆_effective / 10
     subsolver_options.ν = ν
+    subsolver_args = subsolver == TRDH ? (SpectralGradient(1 / ν, f.meta.nvar),) : ()
```

Comment:
Does this imply that we can only use TR-TRDH with `SpectralGradient`? 
Also, I think the operator should be preallocated outside the while loop to facilitate the use of in-place subsolvers (once in-place methods are implemented for all the subsolvers in the same manner as `R2!`)

Reply:
Yes, for the moment, it’s the only option (the other two don’t perform well anyways). I plan to make this an option when I implement the solver object, so I’m not making an effort to preallocate at this point.

Reply:
The others perform well on some problems, like SVM for example. I'll keep a branch on my fork then if want still want to be able to produce the results of indef-pg.

Reply:
I’m planning to work on the solvers soon, so the option should be in place in the near future.

Reply:
I added a note about that in #129.